html/js/tree/Node.js

```diff
@@ -0,0 +1,118 @@
+GAMBIT.TREE = (function (parentModule) {
+    "use strict";
+
+    /**
+    * Creates a new Node.
+    * @class
+    * @param {Node} [parent] Parent node. If null, this is root.
+    */
+    function Node(parent) {
+        this.parent = parent;
+        this.children = [];
+
+        if (parent === null) { // If this is root set level to 0
+            this.level = 0;
+        } else {
+            parent.addChild(this);
+            this.level = parent.level + 1;
+        }
+    }
+
+    /**
+    * ToString function
+    */
+    Node.prototype.toString = function nodeToString() {
+        return "Node: " + "children.length: " + this.children.length + "; level: " + this.level;
+    };
+
+    /**
+    * Function that draws the node in the global canvas
+    */
+    Node.prototype.draw = function () {
+        var thisNode = this;
+        var circle = GAMBIT.canvas.circle(GAMBIT.CONSTANTS.CIRCLE_SIZE)
+            .addClass('node')
+            .x(this.x)
+            .y(this.y)
+            .click(function() {
+                thisNode.onClick();
+            });
+        // console.log("Drawing at y " + this.level*50 + " and x " + this.x);
+    };
+
+    /**
+    * Function that defines the behaviour of the node on click
+    */
+    Node.prototype.onClick = function () {
+        if (GAMBIT.MODE === GAMBIT.MODES.ADD){
+            if (this.children.length === 0) {
+                // Always start with two nodes
+                GAMBIT.tree.addChildNodeTo(this);
+            }
+            GAMBIT.tree.addChildNodeTo(this);
+        } else {
+            GAMBIT.tree.deleteNode(this);
+        }
+        // Tell the tree to redraw itself
+        GAMBIT.tree.draw();
+    };
+
+
+    /**
+    * Function that adds child to node
+    * @param {Node} node Node to add as child
+    */
+    Node.prototype.addChild = function (node) {
+        this.children.push(node);
+    };
+
+    /**
+    * Function that removes child node from children
+    * @param {Node} node Child node to remove
+    */
+    Node.prototype.deleteChild = function (nodeToDelete) {
```

Comment:
I am thinking it may be better to name this removeChild instead of deleteChild, because here we are not really deleting the node, just removing it from the children list, right?